gradio/chat_interface.py

```diff
@@ -245,8 +245,14 @@ def __init__(
 
     def _setup_events(self) -> None:
         submit_fn = self._stream_fn if self.is_generator else self._submit_fn
+        submit_triggers = (
+            [self.textbox.submit, self.submit_btn.click]
+            if self.submit_btn
+            else [self.textbox.submit]
+        )
         submit_event = (
-            self.textbox.submit(
+            on(
```

Comment:
I think we're triggering an event twice here. 

Run `chat_interface_streaming_echo` on main and on 3.44.2. Clear the network panel and then click the submit button.

Dependency with id 8 is being run twice.

### This branch
<img width="1442" alt="image" src="https://github.com/gradio-app/gradio/assets/41651716/177124f3-3ef3-4d81-9fcf-9f0d38730314">


### 3.44.2


<img width="1442" alt="image" src="https://github.com/gradio-app/gradio/assets/41651716/9b8b17b1-aead-4055-998b-8b97d9943b32">


In the case of the chatinterface this is not a noticeable difference but depending on the root cause it may cause expensive events to be re-triggered.

Reply:
good catch! was an issue specific to chat interface implementation, fixed.